crates/core_arch/src/macros.rs

```diff
@@ -2,38 +2,59 @@
 
 // Helper struct used to trigger const eval errors when the const generic immediate value `imm` is
 // out of `bits`-bit range.
-pub(crate) struct ValidateConstImm<const IMM: i32, const BITS: i32>;
-impl<const IMM: i32, const BITS: i32> ValidateConstImm<IMM, BITS> {
+pub(crate) struct ValidateConstImm<const IMM: i32, const MIN: i32, const MAX: i32>;
+impl<const IMM: i32, const MIN: i32, const MAX: i32> ValidateConstImm<IMM, MIN, MAX> {
     pub(crate) const VALID: () = {
-        let _ = 1 / ((IMM >= 0 && IMM < (1 << BITS)) as usize);
+        let _ = 1 / ((IMM >= MIN && IMM < MAX) as usize);
```

Comment:
```suggestion
        let _ = 1 / ((IMM >= MIN && IMM <= MAX) as usize);
```

I think it will be clearer if `MAX` refers to the inclusive upper bound of the value.